src/Uno.UI.Composition/Composition/CompositionSpriteShape.skia.cs

```diff
@@ -141,9 +131,63 @@ private static SKPaint TryCreateAndClearPaint(in Visual.PaintingSession session,
 				}
 			}
 
-			paint.ColorFilter = session.Filters.OpacityColorFilter;
-
 			return paint;
 		}
+
+		private protected override void OnPropertyChangedCore(string? propertyName, bool isSubPropertyChange)
+		{
+			base.OnPropertyChangedCore(propertyName, isSubPropertyChange);
+
+			switch (propertyName)
+			{
+				case nameof(Geometry) or nameof(CombinedTransformMatrix):
+					if (Geometry?.BuildGeometry() is SkiaGeometrySource2D { Geometry: { } geometry })
+					{
+						var transform = CombinedTransformMatrix;
+						SKPath geometryWithTransformations;
+						if (transform.IsIdentity)
+						{
+							geometryWithTransformations = geometry;
+						}
+						else
+						{
+							geometryWithTransformations = new SKPath();
+							geometry.Transform(transform.ToSKMatrix(), geometryWithTransformations);
```

Comment:
The signature of this method is weird ... it is something new? If so, either make the `geometryWithTransformations` a `ref` parameter, either make the method to return a new instance.

Reply:
Not new. It's from SkiaSharp, not something we added. I agree, it looks really weird, but there is no overload that returns a new instance. There is an overload that takes 1 paramter (the matrix) that makes the modification in place, so I can't even add an extension method that returns a new instance.